File: hw1/src/test/java/MainTest.java

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class MainTest {
    private ByteArrayOutputStream os = new ByteArrayOutputStream();

    @Before
    public void setUpStreams() {
        System.setOut(new PrintStream(os));
    }

    @Test
    public void mainTest() {
        Main.main(new String[]{"5", "-", "2"});
        Assert.assertEquals("Result is: 3", os.toString());
    }
}

File: hw5/src/test/kotlin/MainTest.kt
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import java.io.ByteArrayOutputStream
import java.io.PrintStream

class MainTest {
    private val os = ByteArrayOutputStream()

    @Before
    fun setUpStreams() {
        System.setOut(PrintStream(os))
    }

    @Test
    fun mainTest() {
        Main().main(arrayOf("5", "-", "2"))
        Assert.assertEquals("Result is: 3", os.toString())
    }
}